SofaKernel/modules/SofaEngine/BoxROI.inl

```diff
@@ -613,9 +616,6 @@ void BoxROI<DataTypes>::doUpdate()
         return ;
     }
 
-    if(!d_doUpdate.getValue()){
-        return ;
-    }
     if (d_X0.getValue().size() == 0)
```

Comment:
I can understand the intention that "if the inputs are dirty and doUpdate is false, all outputs are cleared". However, the codes seems to me looks like "No matter inputs are dirty or not, as long as doUpdate is false, all outputs are cleared". This sounds not quite logical.

The reason I want to use doUpdate=0 is related to the problem I proposed in the forum as: https://www.sofa-framework.org/community/forum/topic/a-question-on-applying-surface-pressure-on-body/
In point 2, I'd like to move the acutator I modelled. The way to do this is build up a BoxROI to extract some indices and move the actuator by moving the rest position using RestShapeSpringsForceField. If I use doUpdate, the extracted indices will change.

I understand one way is to set recompute_indices='false' in RestShapeSpringsForceField. However, this is not the default value and users probably will not notice this. Besides, this will hide the problem which we discussed here.

The reason I think this is a bug instead of a feature, is that this only happens when BoxROI work together with RestShapeSpringsForceField, and only when we move the rest position. If we don't move the rest position, everything is fine. While when we move the rest position (not sure whether moving rest position is considered as dirty inputs?), the indices in BoxROI will be cleared, and by default all the indices in the model will be constrained in RestShapeSpringsForceField without any warning.

I understand that this section of code may relate to some other code's logic in a big picture of this software which I'm not aware currently.  If this is the case, maybe add a warning message here is a better choice. Please kindly help to advise. Thanks.

Reply:
Thanks a lot for your kind comments and suggestions.